src/obstacle_avoidance/src/MyNode-devel.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+ * Created By: Chris Heathe
+ * Created On:  October 28, 2017
+ * Description: A node that will theoretically interpret lidar data,
+ *              and help Olaf navigate through a hole in a row of cones
+ *              Stress on *theoretically*
+ */
+
+#include <MyNode.h>
+
+MyClass::MyClass(int argc, char **argv, std::string node_name) {
+    // Setup NodeHandles
+    ros::init(argc, argv, node_name);
+    ros::NodeHandle nh;
+    ros::NodeHandle private_nh("~");
+
+    // Obtains character from the parameter server (or launch file), sets '!' as default
+    std::string parameter_name = "my_node/character";
+    std::string default_character = "!";
+    SB_getParam(nh, parameter_name, suffix, default_character);
+
+    // Setup Subscriber(s)
+    std::string topic_to_subscribe_to = "/scan";
+    int refresh_rate = 10;
+    my_subscriber = nh.subscribe(topic_to_subscribe_to, refresh_rate, &MyClass::subscriberCallBack, this);
+
+    // Setup Publisher(s)
+    std::string topic = private_nh.resolveName("/cmd_vel");
+    uint32_t queue_size = 1;
+    my_publisher = private_nh.advertise<geometry_msgs::Twist>(topic, queue_size);
+}
+
+void MyClass::subscriberCallBack(const sensor_msgs::LaserScan::ConstPtr& msg) {
+    std::vector <float> how_far = msg-> ranges;
+    auto scan = *msg;
+    for (int i = 217; i < (how_far.size() - 170); ++i) {
+        //if (how_far[i] < msg->range_min || how_far[i] > msg->range_max || how_far[i] == NAN){
```

Comment:
In general, try to delete commented out code before pushing back to upstream. One of the joys of `git` is that we can always recover old work later (nothing that is commited is ever *really* deleted)